### PR TITLE
Lforms grid view

### PR DESCRIPTION
**Pull request by nikhil-ht**
**Overview**

- Added grid view support to the lforms lib

**How it was tested**

- Tested by creating the local package and use its in sapphire-app

- Tested is auto-fill functionality is worked or not

**Screenshot**

image
https://github.com/lhncbc/lforms/assets/92708869/ed42cfbc-c7d6-4c78-b19b-2d2de549f22e)

updated screenshot
image

### CODE REVIEW

File: src/app/lhc-group-grid/lhc-group-grid.component.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  }
+ 
+  createCustomHeaders(item):void{
+    item.items.forEach(questionItems => {
+      questionItems.items.forEach(subItems => {
+        if(!this.uniqueGridHeaders.includes(subItems.question)  && subItems.skipLogic?.conditions[0]?.trigger?.value.text != 'alwaysHide' &&  subItems.skipLogic?.conditions[0]?.trigger?.value.text != 'Never'){
+          this.uniqueGridHeaders.push(subItems.question);
+        }   
+    });
+    });
```
Comment (semicolin) [inline src/app/lhc-group-grid/lhc-group-grid.component.ts:28]: please format the code

**Review by semicolin — CHANGES_REQUESTED** [review]
In the screenshot there is an extra horizontal line above every form field. I think this is because `lhc-item-question` component applies its own top border in its `:host` style. If we are using table borders instead, we should remove the top border from `lhc-item-question`. I think we can override it from the grid component:
```
lhc-item-question.lhc-de {
    border: 0;
}
```

File: src/app/lhc-group-grid/lhc-group-grid.component.ts (hunk 1 above)
Comment (nikhil-ht) [reply src/app/lhc-group-grid/lhc-group-grid.component.ts:28]: Okay @semicolin  will format the code

**Review by yashpmohod — APPROVED** [review]

**Review by amolht — APPROVED** [review]

File: src/app/lhc-group-grid/lhc-group-grid.component.ts (hunk 1 above)
Comment (nikhil-ht) [reply src/app/lhc-group-grid/lhc-group-grid.component.ts:28]: @semicolin    I have formatted the code made a changes that you suggested .

**Review by semicolin — APPROVED** [review]

**Pull request merged**